File: Back-end (ISS)/Server-Back-ISS/src/main/java/ZgazeniSendvic/Server_Back_ISS/dto/RouteDTO.java
package ZgazeniSendvic.Server_Back_ISS.dto;

import ZgazeniSendvic.Server_Back_ISS.model.Location;
import java.util.List;

public class RouteDTO {
    public Long id;
    public Location start;
    public Location destination;
    public List<Location> midPoints;
}

File: Web Front-end (IKS)/Web-Front-IKS/src/app/models/route.dto.ts
export interface Location {
  latitude: number;
  longitude: number;
  address: string;
}

export interface Route {
  id: number;
  locations: Location[];
}

export interface LocationDTO {
  latitude: number;
  longitude: number;
}

export interface RouteDTO {
  id: number;
  start: LocationDTO;
  destination: LocationDTO;
  midPoints: LocationDTO[];
}